packages/features/ee/workflows/api/scheduleEmailReminders.ts

```diff
@@ -214,190 +256,183 @@ async function handler(req: NextApiRequest, res: NextApiResponse) {
-                mailSettings: {
-                  sandboxMode: {
-                    enable: sandboxMode,
-                  },
+      if (emailContent.emailSubject.length > 0 && !emailBodyEmpty && sendTo) {
+        const batchIdResponse = await client.request({
+          url: "/v3/mail/batch",
+          method: "POST",
+        });
+
+        const batchId = batchIdResponse[1].batch_id;
+
+        if (reminder.workflowStep.action !== WorkflowActions.EMAIL_ADDRESS) {
+          await sendEmailPromises.push(
+            sgMail.send({
+              to: sendTo,
+              from: {
+                email: senderEmail,
+                name: reminder.workflowStep.sender || "Cal.com",
+              },
+              subject: emailContent.emailSubject,
+              html: emailContent.emailBody,
+              batchId: batchId,
+              sendAt: dayjs(reminder.scheduledDate).unix(),
+              replyTo: reminder.booking.user?.email || senderEmail,
+              mailSettings: {
+                sandboxMode: {
+                  enable: sandboxMode,
                 },
-                attachments: reminder.workflowStep.includeCalendarEvent
-                  ? [
-                      {
-                        content: Buffer.from(getiCalEventAsString(reminder.booking) || "").toString("base64"),
-                        filename: "event.ics",
-                        type: "text/calendar; method=REQUEST",
-                        disposition: "attachment",
-                        contentId: uuidv4(),
-                      },
-                    ]
-                  : undefined,
-              })
-            );
-          }
-
-          await prisma.workflowReminder.update({
-            where: {
-              id: reminder.id,
-            },
-            data: {
-              scheduled: true,
-              referenceId: batchId,
-            },
-          });
+              },
+              attachments: reminder.workflowStep.includeCalendarEvent
+                ? [
+                    {
+                      content: Buffer.from(getiCalEventAsString(reminder.booking) || "").toString("base64"),
+                      filename: "event.ics",
+                      type: "text/calendar; method=REQUEST",
+                      disposition: "attachment",
+                      contentId: uuidv4(),
+                    },
+                  ]
+                : undefined,
+            })
+          );
         }
-      } catch (error) {
-        console.log(`Error scheduling Email with error ${error}`);
+
+        await prisma.workflowReminder.update({
+          where: {
+            id: reminder.id,
+          },
+          data: {
+            scheduled: true,
+            referenceId: batchId,
+          },
+        });
       }
+    } catch (error) {
+      logger.error(`Error scheduling Email with error ${error}`);
     }
-    pageNumber++;
   }
 
-  Promise.allSettled(sendEmailPromises).then((results) => {
```

Comment:
Do we intentionally want to fail when there is a failure in sending even one email?

Reply:
Not sure why I removed that 🤔 But we definitely don't want to fail when the sending of one email fails